chef/automate
# Ensure Automate starts when the OS is ready for it to start

After observing a Customer system failure where Automate's hab sup started before the `/hab` filesystem was mounted and discussing internally...

Signed-off-by: Irving Popovetsky <irving@chef.io>

### :nut_and_bolt: Description: What code changed, and why?

This change instructs systemd to start the habitat supervisor after local filesystems (/hab) are mounted and networking is fully initialized. Also inform users not to modify this file, letting them know where customizations can go.

Note1: There is some discussion on if we should add `remote-fs.target`.  This may be desired by customers who wish to mount an NFS share for backups (minio data).  The concern is that an unrelated NFS mount failure could then prevent Automate from starting up.  For now we may want to do this only for Chef Automate Cluster installations where we can interact directly with customers on that.  

Note2: I didn't see any tests specific to the contents of `systemdUnitTemplate`.  Any guidance there would be appreciated.


### :chains: Related Resources

- https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/
- https://manpages.ubuntu.com/manpages/bionic/man7/systemd.special.7.html
- https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/system_administrators_guide/sect-managing_services_with_systemd-targets


### :+1: Definition of Done

### :athletic_shoe: How to Build and Test the Change

1.  Create a server with a separate `/hab` partition and install Automate
2. Reboot the server and observe that Automate may not start reliably
3. Add these changes to a custom.conf and observe a difference

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
